r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -15,6 +15,25 @@
 # specific language governing permissions and limitations
 # under the License.
 
+# developer note:
+# when adapting tests from the dplyr package, note that expect_dplyr_equal()
+# works differently from expect_equal(). the first argument is a dplyr
+# expression containing the placeholder `input`, and the second argument
+# is the R data frame to pass in as that input. So for example, adapt a test
+# from dplyr as follows:
+#
+# dplyr:
+# expect_equal(
+#   mutate(data.frame(x = 1, y = 1), z = 1, x = NULL, y = NULL),
+#   data.frame(z = 1)
+# )
+#
+# arrow:
+# expect_dplyr_equal(
+#   input %>% mutate(z = 1, x = NULL, y = NULL) %>% collect(),
+#   data.frame(x = 1, y = 1)
+# )
```

Comment:
Should this go here or go in https://github.com/apache/arrow/blob/ed4f79c77e5d34d5996fa287df0934dd2f93a12a/r/tests/testthat/helper-expectation.R#L73 where we define `expect_dplyr_equal()`?

Reply:
Good idea; that would probably be a better place to put this "developer note" comment